File: leetcode/commonChar.java
package leetcode;
import java.util.*;

public class commonChar {
    public static void main(String[] args) {
        String ar[] = {"cool","lock","cook"};
        commonChar obj = new commonChar();
        List<String> sol = obj.common(ar);

        System.out.println(Arrays.toString(sol.toArray()));
    }
    public List<String> common(String[] words){
        List<String> ans = new ArrayList<String>();
        int [] dic = new int[26];
        for (int i=0;i<words[0].length();i++){
            dic[words[0].charAt(i) - 'a']++;
        }

        for (int i=1;i<words.length;i++){
            int [] curr = new int[26];
            for(int j=0; j<words[i].length();j++){
                curr[words[i].charAt(j) - 'a']++;
            }

            for(int j=0;j<26;j++){
                if(curr[j]<dic[j]) dic[j] = curr[j];
            }
        }
        for(int i =0;i<25;i++){
            for(int j=0;j<dic[i];j++){
                ans.add(Character.toString((char) ('a'+i)));
            }
        }
        return ans;

    }
}

File: leetcode/commonChar.cpp
#include <iostream>
#include <vector>

using namespace std;

class commonChar
{
public:
    vector<string> commonC(vector<string>& words){
        vector<int> val1(26,0);
        vector<int> val2(26,0);

        for(auto ch : words[0]){
            val1 [ch -'a']++;
        }

        for(int i=1;i<words.size();i++){
            for (auto ch : words[i]){
                val2[ch-'a']++;
            }
            for (int i = 0; i < 26; i++)
            {
                val1[i] = min(val1[i],val2[i]);
                val2[i] = 0;
            }
        }
        vector<string> ans;
        for (int i = 0; i < 26; i++)
        {
            if(val1[i] >0){
                int count = val1[i];
                while(count--){
                    char x = i+ 'a';
                    string s;
                    s =x;
                    ans.push_back(s);
                }
            }
        }
        return ans;
    }
    
};




